### PR TITLE
Implement dl.JSONObject structure

### PR DESCRIPTION
**Pull request by supl**
This PR implements dl.JSONObject structure to represent JavaScript object (generally, map[string]interface{} in Go)

### CODE REVIEW

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Comment by supl** [comment]
> LGTM! Thank you!

@thongdk8 
Thong, I am sorry that I decided to not use the null pattern for JSONObject.
It means that even FromJSON fails to parse a JSON string, it will return an error BUT also not-nil empty JSONObject.

Please check this
https://github.com/scalar-labs/scalardl-go-client-sdk/pull/10/commits/b23b0e88f85a9350bff0e89b491257f1f135156c

I have these changes to simplify the code logic.

**Comment by supl** [comment]
> > LGTM! Thank you!
> 
> @thongdk8 Thong, I am sorry that I decided to not use the null pattern for JSONObject. It means that even FromJSON fails to parse a JSON string, it will return an error BUT also not-nil empty JSONObject.
> 
> Please check this [b23b0e8](https://github.com/scalar-labs/scalardl-go-client-sdk/commit/b23b0e88f85a9350bff0e89b491257f1f135156c)
> 
> I have these changes to simplify the code logic.

Sorry, I think again.
Maybe just keep the null pattern.
I will revert the code.
https://github.com/scalar-labs/scalardl-go-client-sdk/pull/10/commits/55eeb37915b6ee0d1e90de3973398ddd3d9db754

File: json_object.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/google/uuid"
+)
+
+// JSONObject is a general structure to represent JavaScript objects.
+type JSONObject map[string]interface{}
```
Comment (feeblefakie) [inline json_object.go:11]: Just wondering. Is it OK to use `JSON` from the style perspective?
(do we have google go style guide or something else?)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Just left one question.

File: json_object.go (hunk 1 above)
Comment (supl) [reply json_object.go:11]: @feeblefakie 
Yes.
It's ok.

It's mentioned and recommended in
https://github.com/golang/go/wiki/CodeReviewComments#initialisms
The acronyms part in namings should be either in all capitals or in all lower case.

○ userID
○ userid
× userId

○ HTTPServer (for exporting)
○ httpServer (for use inside the package)
× HttpServer

In our case, JSONObject will be exported to the other packages, so the JSONObject naming is fine.
(if it is used internally, then the jsonObject naming would be adopted)

**Comment by supl** [comment]
I will just merge this PR.

**Pull request merged**